sklearn/neighbors/classification.py

```diff
@@ -161,7 +161,7 @@ def predict(self, X):
 
             for k, classes_k in enumerate(classes_):
                 mode = self._mode(_y[neigh_ind, k].toarray(), weights)
-                y_pred_sparse_multilabel.append(csr_matrix(mode).T)
+                y_pred_sparse_multilabel.append(csc_matrix(mode).T)
```

Comment:
This isn't right. You don't want to transpose the CSC. You want to build a CSC on the reshaped mode. Transposing on a numpy array is very cheap. Transposing a scipy.sparse matrix is more expensive (except for COO format).